backend/src/flotilla/services/request_handler.py

```diff
@@ -0,0 +1,166 @@
+from typing import Any, Optional
+
+import requests
+from requests.exceptions import (
+    ConnectionError,
+    ConnectTimeout,
+    HTTPError,
+    RequestException,
+    Timeout,
+)
+from requests.models import Response
+
+from flotilla.settings import settings
+
+
+class RequestHandler:
+    def __init__(self):
+        pass
+
+    def base_request(
+        self,
+        url: str,
+        method: str,
+        json_body: Any,
+        timeout: float,
+        auth: tuple,
+        headers: Optional[dict] = None,
+        data: Optional[dict] = None,
+        params: Optional[dict] = None,
+        **kwargs,
+    ) -> Response:
+        try:
+            response = requests.request(
+                url=url,
+                method=method,
+                auth=auth,
+                headers=headers,
+                timeout=timeout,
+                json=json_body,
+                data=data,
+                params=params,
+                **kwargs,
+            )
+        except Timeout as e:
+            raise RequestException from e
+        except ConnectionError as e:
+            raise RequestException from e
+        except ConnectTimeout as e:
+            raise RequestException from e
+        except RequestException as e:
+            raise RequestException from e
+        except Exception as e:
+            raise RequestException from e
+        try:
+            response.raise_for_status()
+        except HTTPError as e:
+            raise RequestException from e
+        return response
+
+    def get(
+        self,
+        url: str,
+        json_body=None,
+        request_timeout: float = settings.REQUEST_TIMEOUT,
+        auth: Optional[tuple] = None,
+        headers: Optional[dict] = None,
+        data: Optional[dict] = None,
+        params: Optional[dict] = None,
+        **kwargs,
```

Comment:
I see we already use `**kwargs`, why not just use that for all?
```suggestion
        **kwargs,
```

Reply:
It is nice to see the different available parameters when you want to use the function.